processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -60,7 +60,9 @@ public TopNMapFn(
       return resultBuilder.build();
     }
     finally {
-      topNAlgorithm.cleanup(params);
+      if (params != null) {
```

Comment:
any uncaught exception in topNAlgorithm.makeInitParams can cause this, 
In my particular case it was most probably null due jvm being unable to create buffer pool as there were OOMEs at other places , the exact cause of the error was hidden by the NPE stack trace. 

We also have similar check in AggregateTopNMetricFirstAlgo.run() method.